Geometry/src/LUXEGeometry.cpp

```diff
@@ -1,125 +1,112 @@
+    kdtDOpt.range = Acts::RangeXD<3, Acts::ActsScalar>();
+    kdtDOpt.range[0].set(zMin, zMax);
+    kdtDOpt.binningValues = {Acts::BinningValue::binZ};
 
+    auto SP = std::make_shared<Acts::Experimental::Geant4SurfaceProvider<3>>(
+            spCfg, kdtDOpt, false);
+
+    auto lbCfg = Acts::Experimental::LayerStructureBuilder::Config();
+    lbCfg.surfacesProvider = SP;
+    auto LB =
+            std::make_shared<Acts::Experimental::LayerStructureBuilder>(lbCfg);
+    return LB;
+}
 std::shared_ptr<const Acts::Experimental::Detector> 
     buildLUXEDetector(std::string gdmlPath, 
         std::vector<std::string> names, 
         Acts::GeometryContext gctx) {
             // Default template parameters are fine
             // when using names as identifiers
-            auto spFullCfg = Acts::Experimental::Geant4SurfaceProvider<>::Config();
-            spFullCfg.gdmlPath = gdmlPath;
-            spFullCfg.surfacePreselector =
-                std::make_shared<Acts::Geant4PhysicalVolumeSelectors::NameSelector>(names,
-                    true);
-    
-            auto spFull = std::make_shared<Acts::Experimental::Geant4SurfaceProvider<>>(
-                spFullCfg, Acts::Experimental::Geant4SurfaceProvider<>::kdtOptions(),
-                false);
-    
-            auto lbFullCfg = Acts::Experimental::LayerStructureBuilder::Config();
-            lbFullCfg.surfacesProvider = spFull;
-    
-            auto lbFull =
-                std::make_shared<Acts::Experimental::LayerStructureBuilder>(lbFullCfg);
-    
-            auto [sFull, vFull, suFull, vuFull] = lbFull->construct(gctx);
-    
-            assert(sFull.size() == names.size());
+            auto layerBuilder1 = makeLayerBuilder(gdmlPath, names, gctx, 3942, 3972);
+            auto layerBuilder2 = makeLayerBuilder(gdmlPath, names, gctx, 4042, 4072);
+            auto layerBuilder3 = makeLayerBuilder(gdmlPath, names, gctx, 4142, 4172);
+            auto layerBuilder4 = makeLayerBuilder(gdmlPath, names, gctx, 4242, 4272);
 
             // Create the detector
-            auto bounds = std::make_unique<Acts::CuboidVolumeBounds>(3, 3, 3);
-            auto volume = Acts::Experimental::DetectorVolumeFactory::construct(
-                Acts::Experimental::defaultPortalAndSubPortalGenerator(), gctx,
-                "volume", Acts::Transform3::Identity(), std::move(bounds), {}, {},
-                Acts::Experimental::tryNoVolumes(),
-                Acts::Experimental::tryAllPortalsAndSurfaces());
-            volume->assignGeometryId(1);
-            auto detector = Acts::Experimental::Detector::makeShared(
-                "detector", {volume}, Acts::Experimental::tryRootVolumes());
-            
+            std::vector<Acts::BinningValue> detectorBins = {Acts::binZ};
+            std::vector<Acts::ActsScalar> detectorBounds = {100, 100, 5000};
+
+            // The root node - detector
+            auto positronArmBpr = std::make_unique<Acts::Experimental::Blueprint::Node>(
```

Comment:
Let's move the BP construction into a separate function. The builder pipeline is general enough to take whatever BP you have and run with it